core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,625 @@
+      mondoo-m365-ensure-multifactor-authentication-is-enabled-for-all-users-in-administrative-roles: null
+      mondoo-m365-ensure-that-between-two-and-four-global-admins-are-designated: null
+      mondoo-m365-enable-conditional-access-policies-to-block-legacy-authentication: null
+      mondoo-m365-ensure-security-defaults-is-disabled-on-azure-active-directory: null
+      mondoo-m365-ensure-that-spf-records-are-published-for-all-exchange-domains: null
+      mondoo-m365-ensure-that-mobile-devices-require-a-minimum-password-length-to-prevent-brute-force-attacks: null
+      mondoo-m365-ensure-that-mobile-device-encryption-is-enabled-to-prevent-unauthorized-access-to-mobile-data: null
+      mondoo-m365-ensure-third-party-integrated-applications-are-not-allowed: null
+      mondoo-m365-enable-azure-ad-identity-protection-sign-in-risk-policies: null
+      mondoo-m365-enable-azure-ad-identity-protection-user-risk-policies: null
+queries: 
+  - uid: mondoo-m365-ensure-multifactor-authentication-is-enabled-for-all-users-in-all-roles
+    title: Ensure that multifactor authentication has been enabled for all users
+    severity: 80
+    docs:
+      desc: |
+        This control checks that the MFA has been enabled for all users in the Microsoft 365 tenant. 
+      audit: |
+        __cnquery run__
+
+        To audit Microsoft MS365 with `cnquery run`:
+
+        Run the following query:
+
+          ```mql
+          cnquery run ms365 -c "msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'MFARegistrationV2' &&  _['score'] == 9)" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'MFARegistrationV2' &&  _['score'] == 9 )
+          ```
+      remediation: |
+        ### MS365 Console
+
+        To update via the ms365 portal:
+
+        1. Log in as Global Administrator to the ms365 portal at https://admin.microsoft.com
+        2. Click on "Azure Active Directory" once in the "Admin Centers"
+        3. Applications --> Enterprise applications --> Security --> Conditional Access 
+        4. Make sure there is a Policy with following conditions: 
+           * Grant --> Grant access (Require multifactor authentication)
+           * Users --> All users
+           * Cloud apps or actions --> All cloud apps 
+        5. Make sure that the policy is Enabled 
+ 
+    query: |
+      msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'MFARegistrationV2' &&  _['score'] == 9)
```

Comment:
`msgraph.beta` is deprecated. This should be `microsoft.security`

Reply:
Thanks Chris, done!